library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcPOJOInputOperator.java

```diff
@@ -0,0 +1,582 @@
+            break;
+
+          case Types.BIGINT:
+            long longVal = result.getLong(i + 1);
+            ((PojoUtils.SetterLong<Object>) afi.setterOrGetter).set(obj, longVal);
+            break;
+
+          case Types.FLOAT:
+            float floatVal = result.getFloat(i + 1);
+            ((PojoUtils.SetterFloat<Object>) afi.setterOrGetter).set(obj, floatVal);
+            break;
+
+          case Types.DOUBLE:
+            double doubleVal = result.getDouble(i + 1);
+            ((PojoUtils.SetterDouble<Object>) afi.setterOrGetter).set(obj, doubleVal);
+            break;
+
+          case Types.DECIMAL:
+            BigDecimal bdVal = result.getBigDecimal(i + 1);
+            ((PojoUtils.Setter<Object, BigDecimal>) afi.setterOrGetter).set(obj, bdVal);
+            break;
+
+          case Types.TIMESTAMP:
+            Timestamp tsVal = result.getTimestamp(i + 1);
+            ((PojoUtils.SetterLong<Object>) afi.setterOrGetter).set(obj, tsVal.getTime());
+            break;
+
+          case Types.TIME:
+            Time timeVal = result.getTime(i + 1);
+            ((PojoUtils.SetterLong<Object>) afi.setterOrGetter).set(obj, timeVal.getTime());
+            break;
+
+          case Types.DATE:
+            Date dateVal = result.getDate(i + 1);
+            ((PojoUtils.SetterLong<Object>) afi.setterOrGetter).set(obj, dateVal.getTime());
+            break;
+
+          default:
+            handleUnknownDataType(type, obj, afi);
+            break;
+        }
+      }
+      return obj;
+    } catch (SQLException e) {
+      store.disconnect();
+      throw new RuntimeException("fetching metadata", e);
+    }
+  }
+
+  @SuppressWarnings("UnusedParameters")
+  protected void handleUnknownDataType(int type, Object tuple, ActiveFieldInfo activeFieldInfo)
+  {
+    throw new RuntimeException("unsupported data type " + type);
+  }
+
+  @Override
+  public String queryToRetrieveData()
+  {
+    StringBuilder builder = new StringBuilder();
+
+    if (query != null) {
+      builder.append(query.trim());
+      if (builder.charAt(builder.length() - 1) == ';') {
+        builder.deleteCharAt(builder.length() - 1);
+      }
+    } else {
+      builder.append("SELECT ").append(columnsExpression).append(" FROM ").append(tableName);
+      if (whereCondition != null) {
+        builder.append(" WHERE ").append(whereCondition);
+      }
+      if (groupByClause != null) {
+        builder.append(" GROUP BY ").append(groupByClause);
+        if (havingCondition != null) {
+          builder.append(" HAVING ").append(havingCondition);
+        }
+      }
+      if (orderByExpr != null) {
+        builder.append(" ORDER BY ").append(orderByExpr);
+      }
+    }
+    if (limitSupported) {
+      builder.append(" LIMIT ? , ? ");
+    } else {
+      //Starting from Oracle 12c R1 there is a row limiting clause.
+      builder.append(" OFFSET ? ROWS FETCH NEXT ? ROWS FETCH NEXT ? ROWS ONLY ");
```

Comment:
This is also only supported for Oracle, what if we want to make this work with another flavor of SQL? Instead of having limitSupported can we set the limit statement as a string like you are doing for everything else?